### ISSUE TITLE
gps-tracking - must wat for gps signel to start

### ISSUE THREAD

**Issue by mj10777** [geopaparazzi/geopaparazzi#275]
Often when coming out of the U-Bahn, I would like to start the gps-tracking
* but must wait until the gps is properly working

Would it be possible to **note** that the tracking should start when the gps is working?

**Comment by moovida** [comment]
You mean something like: "Gps is currently off, logging will start as soon as the signal is acquired"?
That sounds like a nice thing to have, since you want to start and then put the phone away. It could even be before going out of the house.

But I think this should be optional. One might start and then realize that for any reason the GPS did never gather fix, hence loose some work.

Not trivial to implement. But I will keep this in line as a todo.

**Comment by mj10777** [comment]
But not as a messagebox, just a text message warning.

I was assuming that there is loop checking if gps is on/off
* when that goes on: check for a pending tracking request

**Comment by moovida** [comment]
No, currently when there is not signal it is not possible to start a log. That's it.

I think a messagebox is necessary, so that a user can choose if that is really what he wants.

**Comment by mj10777** [comment]
How does the gps-button turn green when there is suddenly an appropriate signal?

**Comment by moovida** [comment]
>
> How does the gps-button turn green when there is suddenly an appropriate
> signal?
>

There is a listener to the GpsService. A class can register to it and will
get updates with new locations and gps status.

**Comment by brentfraser** [comment]
One thing we could do is make more use of color on the dashboard.  Set the color of the GPS Log tile according to the colors of the GPS Status icon; from the doc:

"The GPS status icon can be of the following colors: RED if the GPS is not switched on, ORANGE if the GPS is on but no fix was acquired, GREEN if the GPS is on and has fix, but no log is being recorded and BLUE if the GPS is logging."

Although that would kind of make it backwards from the green and orange colors used now, but I do like being consistent.

**Comment by moovida** [comment]
@brentfraser , I have to admit this is once more a very cool proposal. I will see if I can test this soon.
